osquery/tables/system/magic.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <stdio.h>
+#include <magic.h>
+
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+static int getMagicData(const int magic_flag, const std::string file_path, std::string &magic_data) {
```

Comment:
Maybe `static bool getMagicData(int magic_flag, const std::string& file_path, std::string& magic_data) {`

Use a bool if you're only returning a yes/no, otherwise a `Status`. And pass the `const std::string` path as a reference.

Be sure to `make format`!